app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -49,6 +50,12 @@ module SmartProxyExtensions
         has_many :hostgroups, :class_name => "::Hostgroup",     :foreign_key => :content_source_id,
                               :inverse_of => :content_source
 
+        belongs_to :content_host,
+                   :class_name => "Katello::System",
+                   :inverse_of => :capsule,
+                   :foreign_key => :content_host_id,
+                   :dependent => :delete
```

Comment:
I would leave this as `:dependent => :nullify`  We don't want to delete the content host if we delete the smart proxy.